### PR TITLE
fix: athena query size limit breaking Upload CSV

### PR DESCRIPTION
**Pull request by a-cid**
### SUMMARY
Athena API has a query size limit of 262144 bytes. Pandas `to_sql` will try to execute an INSERT INTO containing all records, which fails if the file is too large due to this limitation. 

The PyAthena DBAPI provides [helper methods](https://github.com/laughingman7743/PyAthena/#to-sql) that work around this limitation. This PR modifies Athena's db_engine_specs, adding an override to `create_table_from_csv` that defers to PyAthena's `to_csv` helper method when appropriate.

### TEST PLAN
Try uploading CSV file larger than ~250kB

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset/db_engine_specs/athena.py [hunk 1]
```diff
@@ -64,3 +73,37 @@ def _mutate_label(label: str) -> str:
         :return: Conditionally mutated label
         """
         return label.lower()
+
+    @classmethod
+    def create_table_from_csv(  # pylint: disable=too-many-arguments
```
Comment (bkyryliuk) [inline superset/db_engine_specs/athena.py:78]: it seems to me that a cleaner solution would be to move chunksize to the config / make it per engine & set it to lower value for athena
https://github.com/apache/superset/blob/790ac5ef870815978f56ce81b806d43c18421940/superset/views/database/views.py#L193

Comment (a-cid) [reply superset/db_engine_specs/athena.py:78]: Agreed it does seem cleaner and it was my first idea for solving this, but I fear even with a lower chunk size we may still run into problems when trying to upload certain datasets (e.g. CSVs containing long text columns)

pyathena's to_csv converts the dataframe to parquet and uploads it to S3, bypassing this limitation completely

Comment (bkyryliuk) [reply superset/db_engine_specs/athena.py:78]: I see, in that case I would suggest to follow: https://github.com/apache/superset/blob/efeb5a93e62440ddf16f16aa0bd8aeac52eac4d5/superset/db_engine_specs/hive.py#L191

and implement support for various csv controls like null values, custom separator etc

looks like not all of those supported here: https://github.com/laughingman7743/PyAthena/blob/5942373e31668367ddd2fb9fdd3b9c496d388ec7/pyathena/pandas/util.py#L134

Comment (a-cid) [reply superset/db_engine_specs/athena.py:78]: Sorry, I'm confused. Aren't these csv controls already handled by `csv_to_df`?
Are you suggesting to move the csv controls to the create table definition and upload the file as it is, like in the example you linked?

Comment (a-cid) [reply superset/db_engine_specs/athena.py:78]: After having the time to further test the feature I can see this fix clearly needs additional work. Append and Replace are not supported and can corrupt/duplicate your data if used. Will try to do a more proper implementation following @bkyryliuk 's advice.

**Comment by villebro** [comment]
@a-cid I'm closing this as it appears this PR needs to be reimplemented. Let me know if you wish to continue working on this branch and I'll be happy to reopen/rereview it.

**Pull request closed without merging**